app/models/vm_or_template.rb

```diff
@@ -1288,13 +1288,13 @@ def self.folder_entry(ent_desc, cat)
     ent
   end
 
-  def event_where_clause(assoc = :ems_events)
-    case assoc.to_sym
-    when :ems_events, :event_streams
-      return ["vm_or_template_id = ? OR dest_vm_or_template_id = ? ", id, id]
-    when :policy_events
-      return ["target_id = ? and target_class = ? ", id, self.class.base_class.name]
-    end
+  def event_where_clause_ems_events
+    super.or(EmsEvent.where("event_streams.dest_vm_or_template_id" => id))
+  end
+
+  def event_where_clause_miq_events
+    super.or(MiqEvent.where(:target_id => id)
+                     .where(:target_type => self.class.base_class.name))
```

Comment:
can you merge these into a single where clause?
wonder if `.where(:target => self)` is easier to read.